src/javascript/app/pages/user/metatrader/metatrader.ui.js

```diff
@@ -561,6 +564,26 @@ const MetaTraderUI = (() => {
         }
     };
 
+    const setCounterpartyAndJurisdictionTooltip = ($el, acc_type) => {
+        const mt_financial_company = State.getResponse('landing_company.mt_financial_company');
+        const mt_gaming_company = State.getResponse('landing_company.mt_gaming_company');
```

Comment:
this doesn't fix the issue that I mentioned. this function gets called multiple times in `.map`, and it keeps fetching the state response every time. better store it outside the map function.

Reply:
This function is being called outside of the `map()`, I have only found a single instance of `.map()` in the file:
```
            // Update account info
            $detail.find('.acc-info div[data]').map(function () {
                const key     = $(this).attr('data');
                const info    = accounts_info[acc_type].info[key];
                const mapping = {
                    balance : () => (isNaN(info) ? '' : Currency.formatMoney(MetaTraderConfig.getCurrency(acc_type), +info)),
                    leverage: () => `1:${info}`,
                    login   : () => (`${info} (${/demo/.test(accounts_info[acc_type].account_type) ? localize('Demo Account') : localize('Real-Money Account')})`),
                };
                $(this).html(typeof mapping[key] === 'function' ? mapping[key]() : info);
            });

            setCounterpartyAndJurisdictionTooltip($('.acc-info div[data="login"]'), acc_type);
```
I can't find any other instance of `.map()`, maybe I'm misunderstanding.